FirebaseRemoteConfig/SwiftNew/ConfigSettings.swift

```diff
@@ -0,0 +1,579 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import GoogleUtilities
+
+// TODO(ncooke3): Once Obj-C tests are ported, all `public` access modifers can be removed.
+
+private let kRCNGroupPrefix = "frc.group."
+private let kRCNUserDefaultsKeyNamelastETag = "lastETag"
+private let kRCNUserDefaultsKeyNameLastSuccessfulFetchTime = "lastSuccessfulFetchTime"
+private let kRCNAnalyticsFirstOpenTimePropertyName = "_fot"
+private let kRCNExponentialBackoffMinimumInterval = 60 * 2 // 2 mins.
+private let kRCNExponentialBackoffMaximumInterval = 60 * 60 * 4 // 4 hours.
+
+let RCNHTTPDefaultConnectionTimeout: TimeInterval = 60
+
+/// This internal class contains a set of variables that are unique among all the config instances.
+/// It also handles all metadata. This class is not thread safe and does not
+/// inherently allow for synchronized access. Callers are responsible for synchronization
+/// (currently using serial dispatch queues).
+@objc(RCNConfigSettings) open class ConfigSettings: NSObject {
+  // MARK: - Private Properties
+
+  /// A list of successful fetch timestamps in seconds.
+  private var _successFetchTimes: [TimeInterval] = []
```

Comment:
There was a comment here: "/// TODO Not used anymore. Safe to remove."

Reply:
I had this TODO preserved for the ivar's corresponding computed property. Since it was unused, I just removed it and another one in b7e96f9f64206edebae0f64ac85ab594a939d133